### PR TITLE
WIP: updated deps

### PR DESCRIPTION
**Pull request by gilescope**
could just do sha2 if this is too racey.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,12 @@ documentation = "https://docs.rs/substrate-bip39"
...
 
 [dependencies]
-pbkdf2 = { version = "0.8.0", default-features = false }
-sha2 = "0.9.5"
+pbkdf2 = { version = "0.9.0", default-features = false }
+sha2 = "0.9.8"
 hmac = "0.11.0"
-schnorrkel = "0.9.1"
-zeroize = { version = "1.0.0", default-features = false }
+schnorrkel = "0.10.1"
```
Comment (bkchr) [inline Cargo.toml:15]: We don't use that in Substrate. So, we first should check this and make sure nothing breaks.

Comment (bkchr) [reply Cargo.toml:15]: I don't understand your answer.

Comment (bkchr) [reply Cargo.toml:15]: I meant that we don't use schnorrkel 0.10.1 in Substrate while that is already released since quite some time. Before we merge this, we should ensure that we can bump schnorrkel in Substrate as well and nothing breaks.

**Comment by NukeManDan** [comment]
if #17 is done, using workspace defined deps from the SDK would likely resolve any need to keep this updated ind.

**Comment by andresilva** [comment]
Done in https://github.com/paritytech/polkadot-sdk/pull/3579.
